tempto-kafka/src/main/java/io/prestodb/tempto/fulfillment/table/kafka/KafkaTableManager.java

```diff
@@ -0,0 +1,225 @@
+import io.prestodb.tempto.query.QueryResult;
+import kafka.admin.AdminUtils;
+import kafka.admin.RackAwareMode;
+import kafka.utils.ZKStringSerializer$;
+import kafka.utils.ZkUtils;
+import org.I0Itec.zkclient.ZkClient;
+import org.I0Itec.zkclient.ZkConnection;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.slf4j.Logger;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import java.util.Iterator;
+import java.util.Properties;
+import java.util.function.Consumer;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.slf4j.LoggerFactory.getLogger;
+
+@TableManager.Descriptor(tableDefinitionClass = KafkaTableDefinition.class, type = "KAFKA")
+@Singleton
+public class KafkaTableManager
+        implements TableManager<KafkaTableDefinition>
+{
+    private static final Logger LOGGER = getLogger(KafkaTableManager.class);
+
+    private final String databaseName;
+    private final QueryExecutor prestoQueryExecutor;
+    private final String brokerHost;
+    private final Integer brokerPort;
+    private final String prestoKafkaCatalog;
+    private final String zookeeperHost;
+    private final Integer zookeeperPort;
+
+    @Inject
+    public KafkaTableManager(
+            @Named("databaseName") String databaseName,
+            @Named("brokerHost") String brokerHost,
+            @Named("brokerPort") int brokerPort,
+            @Named("zookeeperHost") String zookeeperHost,
+            @Named("zookeeperPort") int zookeeperPort,
+            @Named("prestoDatabaseName") String prestoDatabaseName,
+            @Named("prestoKafkaCatalog") String prestoKafkaCatalog,
+            Injector injector)
+    {
+        this.databaseName = requireNonNull(databaseName, "databaseName is null");
+        this.brokerHost = requireNonNull(brokerHost, "brokerHost is null");
+        this.brokerPort = requireNonNull(brokerPort, "brokerPort is null");
+        this.zookeeperHost = requireNonNull(zookeeperHost, "zookeeperHost is null");
+        this.zookeeperPort = requireNonNull(zookeeperPort, "zookeeperPort is null");
+        requireNonNull(injector, "injector is null");
+        requireNonNull(prestoDatabaseName, "prestoDatabaseName is null");
+        this.prestoQueryExecutor = injector.getInstance(Key.get(QueryExecutor.class, Names.named(prestoDatabaseName)));
+        this.prestoKafkaCatalog = requireNonNull(prestoKafkaCatalog, "prestoKafkaCatalog is null");
+    }
+
+    @Override
+    public TableInstance<KafkaTableDefinition> createImmutable(KafkaTableDefinition tableDefinition, TableHandle tableHandle)
+    {
+        verifyTableExistsInPresto(tableHandle.getSchema().orElseThrow(() -> new IllegalArgumentException("Schema required for Kafka tables")), tableHandle.getName());
+        deleteTopic(tableDefinition.getTopic());
```

Comment:
fail if it already exists?

Reply:
Fail if that does not exist.

Reply:
Hm.. why `createImmutable` contains `deleteTopic`? Is it supposed to exist prior to calling this method?

(can leave as is)

Reply:
Just for sake if it existed before. If we reuse Kafka container for example and  previous tests run were interrupted.